tests/test_s3crypt.py

```diff
@@ -44,3 +46,75 @@ def test_init(self):
         # Run a second time to ensure it is idempotent
         s3crypt.init()
         self.assertEqual(s3crypt.config, self.config_data)
+
+    def test_s3_event_empty(self):
+        s3crypt.init()
+
+        event = {}
+        s3crypt.process_event(event, {})
+
+    def test_s3_event_no_records(self):
+        s3crypt.init()
+
+        event = {'Records': []}
+        s3crypt.process_event(event, {})
+
+
+class TestS3CryptEvent(BaseTest):
+    def test_s3_event_simple(self):
+        self.patch(s3, 'S3_AUGMENT_TABLE', [])
+        session_factory = self.replay_flight_data('test_s3_encrypt')
+        client = session_factory().client('s3')
+        self.patch(s3crypt, 's3', client)
+
+        event = {'Records': [{
+            's3': {
+                'bucket': {
+                    'name': 'test-bucket'
+                },
+                'object': {
+                    'key': 'test-key',
+                    'size': 42
+                }
+            }
+        }]}
+        s3crypt.process_event(event, {})
+
+    def test_s3_event_unsafe_key(self):
+        self.patch(s3, 'S3_AUGMENT_TABLE', [])
+        session_factory = self.replay_flight_data('test_s3_encrypt')
+        client = session_factory().client('s3')
+        self.patch(s3crypt, 's3', client)
+
+        event = {'Records': [{
+            's3': {
+                'bucket': {
+                    'name': 'test-bucket'
+                },
+                'object': {
+                    'key': '/test000/!-_.*\'()/&@:,$=+%2b?;/ /whatever',
```

Comment:
so keys come in via events url encoded, the tests should be checking that. http://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html

Reply:
Indeed.